src/main/host/descriptor/socket/netlink.rs

```diff
@@ -689,8 +689,28 @@ impl InitialState {
             let nlmsg_type = u16::from_ne_bytes(nlmsg_type.try_into().unwrap());
 
             match nlmsg_type {
-                RTM_GETLINK => self.handle_ifinfomsg(common, &packet_buffer[..]),
-                RTM_GETADDR => self.handle_ifaddrmsg(common, &packet_buffer[..]),
+                RTM_GETLINK => {
+                    let min_len =
+                        std::mem::size_of::<nlmsghdr>() + std::mem::size_of::<ifinfomsg>();
+                    // Pad the message if it's too short and update the len field
+                    if packet_buffer.len() < min_len {
+                        packet_buffer.resize(min_len, 0);
+                        packet_buffer[memoffset::span_of!(nlmsghdr, nlmsg_len)]
+                            .copy_from_slice(&(min_len as u32).to_le_bytes()[..]);
+                    }
```

Comment:
I think it's better to only pad the bytes if we have at least an entire `nlmsghdr`. So I think something like:

```rust
let nlmsghdr_len = std::mem::size_of::<nlmsghdr>();
let ifinfomsg_len = std::mem::size_of::<ifinfomsg>();
if (nlmsghdr_len..(nlmsghdr_len + ifinfomsg_len)).contains(packet_buffer.len()) { ... }
```

If this ends up being too restrictive, then we can change it in the future.